src/types/events/message-events.ts

```diff
@@ -0,0 +1,128 @@
+import { MessageAttachment, KnownBlock, Block } from '@slack/types';
+
+export type MessageEvent =
+  | GenericMessageEvent
+  | BotMessageEvent
+  | EKMAccessDeniedMessageEvent
+  | MeMessageEvent
+  | MessageChangedEvent
+  | MessageDeletedEvent
+  | MessageRepliedEvent
+  | ThreadBroadcastMessageEvent;
+
+export interface GenericMessageEvent {
```

Comment:
The type definition seems to be missing some fields such as team and channel_type. You can check the Java SDK's type definition to learn them. https://github.com/slackapi/java-slack-sdk/tree/v1.4.0/slack-api-model/src/main/java/com/slack/api/model/event

Reply:
Great catch! Should I create another issue for this so it can be addressed in a separate PR? This PR was meant to solve the structural issue with the union types, but the issue you've pointed out is independent of that.

Reply:
Let's create a new issue for it!